### ISSUE TITLE
When rsyslog is not installed, missing rsyslog_version fact causes template failures

### ISSUE THREAD

**Issue by sirtaj** [saz/puppet-rsyslog#165]
The custom rsyslog_version fact evaluates to nil when the package is not installed, which leads to
template evaluation errors when comparing versions etc:
```
Error: Could not retrieve catalog from remote server: Error 400 on SERVER: Failed to parse template rsyslog/server-default.conf.erb:
  Filepath: /usr/lib/ruby/vendor_ruby/puppet/parser/functions/template.rb
  Line: 26
  Detail: Failed to parse template rsyslog/server/_default-header.conf.erb:
  Filepath: /usr/lib/ruby/vendor_ruby/puppet/util/package.rb
  Line: 4
  Detail: undefined method `scan' for nil:NilClass

 at /etc/puppet/modules/rsyslog/manifests/server.pp:71 on node puppet-test.localtest
```

While this is easy enough to work around, I'm unsure what the correct solution is to this problem. Even if this module installs rsyslog itself, the fact will still be missing until the next agent run. 

Another option is to force the version to an explicit empty string so that the string operations don't fail, which would be a quicker fix than changing all the places the fact is used.

**Comment by joshbeard** [comment]
This is also an issue where `package_status` is latest and a major version change is made during the Puppet run (e.g. from 7.x to 8.x) where the config format has changes.  Since the template determines which syntax to use based on the fact, a major package update prevents the service from starting because the template has the incorrect syntax.  It corrects itself on a subsequent Puppet run.

**Comment by saz** [comment]
This should be resolved with release 4.0.0
